tests/johnson_alahi_li_2016/test_modules.py

```diff
@@ -0,0 +1,315 @@
+            upsample=upsample,
+        )
+
+        assert isinstance(
+            type(conv_module),
+            type(nn.ConvTranspose2d) if upsample else type(nn.Conv2d),
+        )
+
+        with subtests.test("in_channels"):
+            assert conv_module.in_channels == in_channels
+
+        with subtests.test("out_channels"):
+            assert conv_module.out_channels == out_channels
+
+        with subtests.test("kernel_size"):
+            assert conv_module.kernel_size == to_2d_arg(kernel_size)
+
+        with subtests.test("stride"):
+            assert conv_module.stride == to_2d_arg(stride)
+
+        with subtests.test("padding"):
+            assert conv_module.padding == (1, 1)
+
+        if upsample:
+            with subtests.test("output_padding"):
+                assert conv_module.output_padding == to_2d_arg(0)
+
+
+def test_get_norm(subtests):
+    out_channels = 3
+    for instance_norm in (True, False):
+        with subtests.test(instance_norm=instance_norm):
+            norm_module = modules.get_norm(out_channels, instance_norm=instance_norm)
+
+            assert isinstance(
+                type(norm_module),
+                type(nn.InstanceNorm2d) if instance_norm else type(nn.BatchNorm2d),
+            )
+
+            with subtests.test("out_channels"):
+                assert norm_module.num_features == out_channels
+
+            with subtests.test("eps"):
+                assert norm_module.eps == pytest.approx(1e-5)
+
+            with subtests.test("momentum"):
+                assert norm_module.momentum == pytest.approx(1e-1)
+
+            with subtests.test("affine"):
+                assert norm_module.affine
+
+            with subtests.test("track_running_stats"):
+                assert norm_module.track_running_stats
+
+
+def test_johnson_alahi_li_2016_conv_block(subtests):
+    in_channels = out_channels = 3
+    kernel_size = 3
+    stride = 1
+    for relu in (True, False):
+        with subtests.test(relu=relu):
+            conv_block = modules.johnson_alahi_li_2016_conv_block(
+                in_channels, out_channels, kernel_size, stride=stride, relu=relu
+            )
+
+            assert isinstance(type(conv_block), type(nn.Sequential))
+
+            assert len(conv_block) == 3 if relu else 2
+            assert isinstance(type(conv_block[0]), type(nn.Conv2d))
+            assert isinstance(type(conv_block[1]), type(nn.InstanceNorm2d))
+            if relu:
+                assert isinstance(type(conv_block[2]), type(nn.ReLU))
+                assert conv_block[2].inplace
+
+
+def test_johnson_alahi_li_2016_residual_block(subtests, input_image):
+    channels = 3
+    residual_block = modules.johnson_alahi_li_2016_residual_block(channels)
+
+    assert isinstance(type(residual_block), type(ResidualBlock))
+
+    with subtests.test("residual"):
+        assert isinstance(type(residual_block.residual), type(nn.Sequential))
+        for i in range(0, 2):
```

Comment:
A little overkill to use a loop for 2 elements. I think it is sufficient to check if `len(residual_block.residual) == 2`.